src/com/skillsup/dao/UserInFileDaoImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.skillsup.dao;
+
+import com.skillsup.model.Gender;
+import com.skillsup.model.User;
+import com.skillsup.model.UserRole;
+
+import java.io.*;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class UserInFileDaoImpl implements UserDAO {
+    private static final String PATH_USER_DB_FILE = "E:\\onlineShop\\src\\com\\skillsup\\resources\\userDB.txt";
+    private static final User ADMIN = new User("admin", "admin", UserRole.ADMIN);
+
+    @Override
+    public void saveUser(User user) {
+        try {
+            FileWriter fileWriter = new FileWriter(PATH_USER_DB_FILE, true);
+            String userString = user.getUsername() + "!" + user.getPassword() + "!" +
+                    user.getUserRole() + "!" + user.getGender() + "!" + user.getAge() + "!" + user.getAccount() + "\n";
+            fileWriter.append(userString);
+            fileWriter.flush();
+            fileWriter.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void deleteUser(User user) {
+        Map<String, User> userMap = getUserMapFromFile();
+        userMap.remove(user.getUsername());
+        try {
+            Files.delete(Paths.get(PATH_USER_DB_FILE));
```

Comment:
deleting whole file is not very efficient

Reply:
Interesting, but then how best to do?